datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/resolvers/ingest/source/UpsertIngestionSourceResolver.java

```diff
@@ -119,4 +129,30 @@ private DataHubIngestionSourceSchedule mapSchedule(final UpdateIngestionSourceSc
     result.setTimezone(input.getTimezone());
     return result;
   }
+
+  private String optionallySetPipelineName(String recipe, String ingestionSourceUrnString) {
+    try {
+      JSONObject jsonRecipe = new JSONObject(recipe);
+      boolean isStatefulIngestionEnabled = false;
```

Comment:
Do we need to check all this stuff? Why not just always provide pipeline_name, regardless of whether stateful ingestion is enabled? I don't think it will hurt

Reply:
That would allow us to skip right to line 147

Reply:
oh man that makes me so happy - I also agree it doesn't hurt - I'm gonna skip this stuff cuz it's ugly and annoying

Reply:
the only thing is that if people aren't using forms and don't set stateful_ingestion in their recipe, next time they create a recipe or edit one there will all of a sudden be this `pipeline_name` field on it and they might not know why/ This is super minor and still down to do it, just tossing out there.

Reply:
The other option then is to inject this when an execution request is created (createIngestionExecutionRequest) 

so that we do not change their recipe files. This may actually be preferable